data-prepper-plugins/log-http-source/src/main/java/com/amazon/dataprepper/plugins/source/loghttp/LogThrottlingStrategy.java

```diff
@@ -0,0 +1,48 @@
+/*
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  The OpenSearch Contributors require contributions made to
+ *  this file be licensed under the Apache-2.0 license or a
+ *  compatible open source license.
+ *
+ *  Modifications Copyright OpenSearch Contributors. See
+ *  GitHub history for details.
+ */
+
+package com.amazon.dataprepper.plugins.source.loghttp;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.util.UnmodifiableFuture;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.throttling.ThrottlingStrategy;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CompletionStage;
+
+public class LogThrottlingStrategy extends ThrottlingStrategy<HttpRequest> {
+    private final int maxPendingRequests;
+    private final BlockingQueue<Runnable> queue;
+
+    public LogThrottlingStrategy(int maxPendingRequests, BlockingQueue<Runnable> queue) {
+        this.maxPendingRequests = maxPendingRequests;
+        this.queue = queue;
```

Comment:
add validation

Reply:
What do we want to validate on queue?

Reply:
not null check for queue and min and max value validation for maxPendingRequests

Reply:
Will add nonnull check as a safeguard. But for validation on maxPendingRequests we do need an agreement as mentioned above.